QualificationsRegister/Database/RegisterDb.cs

```diff
@@ -51,30 +52,121 @@ public RegisterDb(RegisterDbContext registerDbContext, ILoggerFactory loggerFact
 
         #region Qualifications Private
 
-        public ListResponse<Qualification> GetQualificationsByName(int page, int limit, string title = "")
+        public ListResponse<Qualification> GetQualificationsByName(int page, int limit, QualificationFilter? query, string title = "")
         {
             var quals = _context.Qualifications.OrderBy(e => e.QualificationNumber);
             var count = 0;
 
+            //to initialise the IQueryable
+            var filteredList = quals.Where(e => e.Id != null);
+
             if (!string.IsNullOrEmpty(title))
             {
-                var filteredList = quals.Where(q => q.Title.Contains(title));
+                filteredList = filteredList.Where(q => q.Title.Contains(title));
+            }
+
+            if (query != null)
+            {
+                if (query.AssessmentMethods != null)
+                {
+                    foreach (var aM in query.AssessmentMethods)
+                    {
+                        filteredList = filteredList.Where(q => q.AssessmentMethods!.Contains(aM));
+                    }
+                }
+
+                if (query.AwardingOrganisations != null)
+                {
+                    foreach (var aO in query.AwardingOrganisations)
+                    {
+                        filteredList = filteredList.Where(q => q.OrganisationName == aO);
+                    }
+                }
+
+                if (query.Availability != null)
+                {
+                    foreach (var av in query.Availability)
+                    {
+                        filteredList = filteredList.Where(q => q.Status == av);
+                    }
+                }
+
+                if (query.QualificationTypes != null)
+                {
+                    foreach (var qT in query.QualificationTypes)
+                    {
+                        filteredList = filteredList.Where(q => q.Type == qT);
+                    }
+                }
+
+                if (query.QualificationLevels != null)
+                {
+                    foreach (var qL in query.QualificationLevels)
+                    {
+                        filteredList = filteredList.Where(q => q.Level == qL);
+                    }
+                }
+
+                if (query.QualificationSubLevels != null)
+                {
+                    foreach (var qSL in query.QualificationSubLevels)
+                    {
```

Comment:
https://stackoverflow.com/questions/51310113/entity-framework-core-linq-query-with-multiple-where-clauses
![image](https://github.com/OfqualGovUK/ofqual-register-api/assets/160255843/9c38a9f4-1c86-422e-b9b0-1354ee9181d9)

Reply:
How would you do the foreach here?